src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -101,16 +117,84 @@ export function ActiveWeb3Provider({
     activating = updateContext.activating ?? false
     active = updateContext.active ?? false
     chainId = updateContext.chainId
-    error = updateContext.error
     library = updateContext.library as Web3Provider
   }
 
-  // Log web3 errors to facilitate debugging.
-  useEffect(() => {
-    if (error) {
-      console.error('web3 error:', error)
+  return <Web3Context.Provider value={{ web3, updateWeb3 }}>{children}</Web3Context.Provider>
+}
+
+export type Web3Connection = [Connector, Web3ReactHooks]
+
+function toWeb3Connection<T extends Connector>([connector, hooks]: [T, Web3ReactHooks, Web3ReactStore]): [
+  T,
+  Web3ReactHooks
+] {
+  return [connector, hooks]
+}
+
+function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: string | JsonRpcProvider) {
+  // TODO(kristiehuang): implement RPC URL fallback, then jsonRpcEndpoint can be optional
+
+  // WalletConnect relies on Buffer, so it must be polyfilled.
+  if (!('Buffer' in window)) {
+    window.Buffer = Buffer
+  }
+
+  let rpcUrl: string
+  if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
+    rpcUrl = jsonRpcEndpoint.connection.url
+  } else {
+    rpcUrl = jsonRpcEndpoint ?? '' // TODO(kristiehuang): use fallback RPC URL
+  }
+
+  return toWeb3Connection(
+    initializeConnector<WalletConnect>(
+      (actions) =>
+        new WalletConnect({
+          actions,
+          options: {
+            rpc: {
+              1: [rpcUrl].filter((url) => url !== undefined && url !== ''),
+            },
+            qrcode: useDefault,
+          }, // TODO(kristiehuang): WC only works on network chainid 1?
+        })
+    )
+  )
+}
+
+function useActiveWalletProvider(): Web3Provider | undefined {
+  return getPriorityConnector(...connections).usePriorityProvider() as Web3Provider
+}
+
+export function useConnect(connection: Web3Connection) {
+  const [wallet, hooks] = connection
+  const isActive = hooks.useIsActive()
+  const accounts = hooks.useAccounts()
+  const account = hooks.useAccount()
+  const activating = hooks.useIsActivating()
+  const chainId = hooks.useChainId()
+  const library = hooks.useProvider()
+
+  const updateActiveWeb3ReactCallback = useUpdateActiveWeb3ReactCallback()
+
+  const useWallet = useCallback(() => {
+    if (!isActive) {
+      wallet.activate()
+    } else {
+      // wallet should be already be active
+      const updateContext: Web3ContextType = {
+        connector: wallet,
+        library,
+        accounts,
+        account,
+        activating,
+        active: isActive,
+        chainId,
+      }
+      updateActiveWeb3ReactCallback(updateContext)
     }
-  }, [error])
+  }, [account, accounts, activating, chainId, isActive, library, updateActiveWeb3ReactCallback, wallet])
 
-  return <Web3Context.Provider value={{ web3, updateWeb3 }}>{children}</Web3Context.Provider>
+  return useWallet
 }
```

Comment:
@zzmp can you take a look at this?

@kristiehuang The name here makes it seem like this is a hook (`use______`), and it's using hooks internally, but the usage in `ConnectWalletDialog` is as a factory for connection handlers.

Reply:
could I just rename it to `useWalletCallback`?

Reply:
Usage is correct, but naming is off (and I think you inherited this).
@kristiehuang could you change `useWallet`->`activateWallet`, and `useConnect`->`useConnectCallback`?